devtools/x/src/context.rs

```diff
@@ -14,16 +15,18 @@ pub struct XContext {
 impl XContext {
     /// Creates a new `GlobalContext` by reading the config in the project root.
     pub fn new() -> Result<Self> {
-        Ok(Self::with_config(Config::from_project_root()?))
+        Self::with_config(Config::from_project_root()?)
     }
 
     /// Creates a new `GlobalContext` based on the given config.
-    pub fn with_config(config: Config) -> Self {
-        Self {
-            core: XCoreContext::new(project_root()),
+    pub fn with_config(config: Config) -> Result<Self> {
```

Comment:
I really like this approach to providing data to a context.   I used similar approaches with spring context for years.

Reply:
Yeah, I was thinking of future testability when I did it that way. Definitely makes it a lot easier.